### PR TITLE
Remove data attributs

### PR DESCRIPTION
**Pull request by itkg-nanne**

### CODE REVIEW

File: Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php [hunk 1]
```diff
@@ -23,17 +24,25 @@
...
      */
-    public function __construct($name, $role, $weight, $parent)
+    public function __construct($name, $role, $weight, $parent, $datatableParameter, $translator)
     {
         $this->name = $name;
         $this->role = $role;
         $this->weight = $weight;
         $this->parent = $parent;
+
+        if (!is_null($translator)) {
```
Comment (nicolasThal) [inline Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php]: whith your constructor signature, `$translator` cannot be null

File: Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php [hunk 2]
```diff
@@ -92,4 +109,21 @@ protected function render($view, array $parameters = array())
...
+     */
+    protected function preFormatDatatableParameter($datatableParameter, TranslatorInterface $translator)
+    {
+        foreach ($datatableParameter as $name => &$parameters) {
+            $parameters['name'] = $name;
+            if (array_key_exists('title', $parameters) && $translator) {
+                $parameters['title'] = $translator->trans($parameters['title']);
+            }
+        }
+        return $datatableParameter;
```
Comment (nicolasThal) [inline Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php]: empty line before return

File: Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php (hunk 1 above)
Comment (itkg-nanne) [inline Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php]: Look at TreeNodesPanelStrategy.php, @nicolasThal

Comment (nicolasThal) [inline Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php]: My purpose was not on the validity of the check, but rather on the method signature, you should have `$translator = null` this way, you are not entitled to put null in the constructor when you call it.

My second point is to modify the check in `$translator instanceof TranslatorInterface`

Comment (itkg-nanne) [inline Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php]: You're right and that's what I did : $datatableParameter = array(), $translator = null. But whith theses default values, I was unable to see which NavigationPanelStrategy was updated with the new logic. So I decided to remove default values to obtain Exception and update strategies. I found it convenient so let it in place. Are you sure, I should add the default values? (ok for the test) @nicolasThal

File: Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php [hunk 3]
```diff
@@ -92,4 +109,21 @@ protected function render($view, array $parameters = array())
     {
         return $this->templating->render($view, $parameters);
     }
+
+    /**
+     * @param array                    $datatableParameter
+     * @param TranslatorInterface $translator
```
Comment (itkg-alavieille) [inline Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php]: align phpdoc

File: Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php [hunk 4]
```diff
@@ -92,4 +109,21 @@ protected function render($view, array $parameters = array())
...
+     *
+     * @return array
+     */
+    protected function preFormatDatatableParameter($datatableParameter, TranslatorInterface $translator)
```
Comment (itkg-ngilain) [inline Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php]: you can type $datatableParameter in the signature

File: Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php [hunk 5]
```diff
@@ -15,27 +17,57 @@ class ContentTypeForContentPanelStrategy extends AbstractNavigationPanelStrategy
...
      */
-    public function __construct(ContentTypeRepositoryInterface $contentTypeRepository, ContextManager $contextManager, $parent, $weight)
+    public function __construct(
+        ContentTypeRepositoryInterface $contentTypeRepository,
+        ContextManager $contextManager,
+        $parent,
+        $weight,
+        $defaultDataParameter,
+        TranslatorInterface $translator,
+        TranslationChoiceManager $translationChoiceManager)
```
Comment (itkg-ngilain) [inline Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php]: the ) should be put on the next line near the {

File: BackofficeBundle/Resources/views/BackOffice/Include/NavigationPanel/Menu/Administration/api_client.html.twig [hunk 6]
```diff
@@ -3,10 +3,7 @@
         id="nav-api_client"
         data-url="{{ path('open_orchestra_api_api_client_list') }}"
         data-add="{{ path('open_orchestra_backoffice_api_client_new') }}"
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/views/BackOffice/Include/NavigationPanel/Menu/Administration/api_client.html.twig]: @itkg-nanne : Can't the data-url and data-add be added to the datatable conf to be removed too ?

File: LogBundle/Resources/views/AdministrationPanel/logs.html.twig [hunk 7]
```diff
@@ -2,11 +2,8 @@
         href="#log/list"
         id="nav-log"
         data-url="{{ path('open_orchestra_api_log_list') }}"
-        data-displayed-elements="date_time, user_ip, user_name, site_name, message"
-        data-visible-elements="date_time, user_ip, user_name, site_name, message"
-        data-input-header = "date, text, text, text, text"
-        data-translated-header="{{'open_orchestra_log.table.log.date_time'|trans }},{{'open_orchestra_log.table.log.user_ip'|trans }},{{'open_orchestra_log.table.log.user_name'|trans }},{{'open_orchestra_log.table.log.site_name'|trans }},{{'open_orchestra_log.table.log.message'|trans }}"
         data-order="0, desc"
```
Comment (itkg-ngilain) [inline LogBundle/Resources/views/AdministrationPanel/logs.html.twig]: @itkg-nanne : can't the data-order be added to the conf to be removed too ?

File: Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php [hunk 8]
```diff
@@ -15,27 +17,57 @@ class ContentTypeForContentPanelStrategy extends AbstractNavigationPanelStrategy
...
+                    $dataParameter[$contentTypeId][$name]['visible'] = true;
+                }
+            }
+            foreach ($fields as $field) {
+                $fieldId = $field->getFieldId();
+                $dataParameter[$contentTypeId][$fieldId] = array(
+                    'name' => 'attributes.' . $fieldId,
+                    'title' => $translationChoiceManager->choose($field->getLabels()),
+                    'visible' => $field->getListable(),
+                    'activateColvis' => true,
```
Comment (itkg-alavieille) [inline Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php]: why by default it's true, i think  you are a problem with visibility of columns, if you look inte and your dev in contents car for instance, in dev all columns are visible and not inte

File: Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php [hunk 9]
```diff
@@ -15,27 +17,57 @@ class ContentTypeForContentPanelStrategy extends AbstractNavigationPanelStrategy
...
+                if (in_array($name, $contentType->getDefaultListable())) {
+                    $dataParameter[$contentTypeId][$name]['visible'] = true;
+                }
+            }
+            foreach ($fields as $field) {
+                $fieldId = $field->getFieldId();
+                $dataParameter[$contentTypeId][$fieldId] = array(
+                    'name' => 'attributes.' . $fieldId,
+                    'title' => $translationChoiceManager->choose($field->getLabels()),
+                    'visible' => $field->getListable(),
```
Comment (itkg-alavieille) [inline Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php]: if fields isn't not listable why add it in the configuration

File: Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php [hunk 10]
```diff
@@ -15,27 +17,57 @@ class ContentTypeForContentPanelStrategy extends AbstractNavigationPanelStrategy
...
+        $defaultDataParameter,
+        TranslatorInterface $translator,
+        TranslationChoiceManager $translationChoiceManager)
     {
-        parent::__construct('content_type_for_content', self::ROLE_ACCESS_CONTENT_TYPE_FOR_CONTENT, $weight, $parent);
-        $this->contentTypeRepository = $contentTypeRepository;
-        $this->contextManager = $contextManager;
+        $dataParameter = array();
+
+        $defaultDataParameter = $this->preFormatDatatableParameter($defaultDataParameter, $translator);
```
Comment (itkg-alavieille) [inline Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php]: i think not is a good idea to add this in constructor, you can move this code in a method

File: BackofficeBundle/Resources/views/BackOffice/Include/NavigationPanel/Menu/Administration/api_client.html.twig (hunk 6 above)
Comment (itkg-nanne) [inline BackofficeBundle/Resources/views/BackOffice/Include/NavigationPanel/Menu/Administration/api_client.html.twig]: not sure.I prefer keeping only datatable configuration in this PR. @itkg-ngilain

**Comment by itkg-alavieille** [comment]
think to add test and update documenation :)

File: Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php (hunk 8 above)
Comment (itkg-nanne) [inline Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php]: activateColvi is actaully always set to true. you can see it in TableviewCollectionView.coffee line 47 @itkg-alavieille

File: Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php (hunk 9 above)
Comment (itkg-nanne) [inline Backoffice/NavigationPanel/Strategies/ContentTypeForContentPanelStrategy.php]: to map between yml and datatable configuration @itkg-alavieille

File: BackofficeBundle/Resources/views/BackOffice/Include/NavigationPanel/Menu/Administration/api_client.html.twig (hunk 6 above)
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/views/BackOffice/Include/NavigationPanel/Menu/Administration/api_client.html.twig]: ok so can you create a ticket to extend the config system ?

**Comment by itkg-nanne** [comment]
Done @itkg-alavieille , @nicolasThal , @itkg-ngilain

File: Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php [hunk 11]
```diff
@@ -23,17 +24,25 @@
...
         $this->name = $name;
         $this->role = $role;
         $this->weight = $weight;
         $this->parent = $parent;
+
+        if ($translator instanceof TranslatorInterface) {
+            $datatableParameter = $this->preFormatDatatableParameter($datatableParameter, $translator);
+        }
+        $this->datatableParameter = $datatableParameter;
+        $this->translator = $translator;
```
Comment (itkg-alavieille) [inline Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php]: why add translator in parameter of class if you don''t use it.
if you add translator in parameter of class, remove it of method preFormatDatatableParameter

File: Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php [hunk 12]
```diff
@@ -92,4 +109,22 @@ protected function render($view, array $parameters = array())
...
+     * @param array               $datatableParameter
+     * @param TranslatorInterface $translator
+     *
+     * @return array
+     */
+    protected function preFormatDatatableParameter(array $datatableParameter, TranslatorInterface $translator)
+    {
+        foreach ($datatableParameter as $name => &$parameters) {
+            $parameters['name'] = $name;
+            if (array_key_exists('title', $parameters) && $translator) {
```
Comment (itkg-alavieille) [inline Backoffice/NavigationPanel/Strategies/AbstractNavigationPanelStrategy.php]: why test $translator exists ? you check already in signature of method if $translator is an instance of TranslatorInterface

File: Backoffice/NavigationPanel/Strategies/AdministrationPanelStrategy.php [hunk 13]
```diff
@@ -57,7 +57,7 @@ class AdministrationPanelStrategy extends AbstractNavigationPanelStrategy
      */
     public function show()
     {
-        return $this->render($this->getTemplate());
+        return $this->render($this->getTemplate(), array('datatableParameterName' => $this->name));
```
Comment (itkg-alavieille) [inline Backoffice/NavigationPanel/Strategies/AdministrationPanelStrategy.php]: why not use method $this->getDatatableParameter ?

**Comment by nicolasThal** [comment]
:+1:

**Comment by itkg-alavieille** [comment]
:+1:

**Pull request merged**